### DISCUSSION TITLE
Conditional rendering components

### DISCUSSION THREAD

**Discussion by davidangelesu** [Kitware/trame#504] [Q&A] [answered]
Hi, is there a way to conditionally render the components depending on the state of the application? 
As an example, i would like to see if it is possible to add some buttons once i change the value of a state variable from:
`state['show_buttons'] = False
`
to:
`state['show_buttons'] = True
`

**Comment by jourdain** [comment]
Yes just use the `v_show="show_buttons"` directive on your buttons.

**Reply by jourdain** [reply, marked as answer]
- https://vuejs.org/guide/essentials/template-syntax.html#directives
- https://vuejs.org/api/built-in-directives.html

**Comment by jourdain** [comment]
Side note, why do you do `state['show_buttons'] = False` instead of `state.show_buttons = False`? The state can support both notation and usually the `[ ]` notation is more useful when the variable name is stored in another variable.

**Comment by davidangelesu** [comment]
You are right. No real reason on my side
